src/iologindata.cpp

```diff
@@ -389,12 +389,7 @@ bool IOLoginData::loadPlayer(Player* player, DBResult_ptr result)
 
 		if (guild) {
 			player->guild = guild;
-			GuildRank* rank = guild->getRankById(playerRankId);
-			if (rank) {
-				player->guildLevel = rank->level;
-			} else {
-				player->guildLevel = 1;
-			}
+			player->guildRank = guild->getRankById(playerRankId);
```

Comment:
guildRank can be null if the rank id of this player was inserted to the database after the first time someone in the guild logged in. The rank should be added to the guild if it does not currently exist.